python_modules/dagster/dagster_tests/definitions_tests/metadata_tests/test_namespaced_metadata_set.py

```diff
@@ -0,0 +1,32 @@
+from typing import Optional
+
+import pytest
+from dagster import FloatMetadataValue, IntMetadataValue, UrlMetadataValue
+from dagster._core.definitions.metadata import NamespacedMetadataSet
+from pydantic import ValidationError
+
+
+def test_extract_primitive_coercion():
+    class MyMetadataSet(NamespacedMetadataSet):
+        primitive_int: Optional[int] = None
+        primitive_float: Optional[float] = None
+        int_metadata_value: Optional[IntMetadataValue] = None
+        url_metadata_value: Optional[UrlMetadataValue] = None
+
+        @classmethod
+        def namespace(cls) -> str:
+            return "foo"
+
+    assert MyMetadataSet.extract({"foo/primitive_int": 5}).primitive_int == 5
+    assert MyMetadataSet.extract({"foo/primitive_float": 5}).primitive_float == 5
+    assert MyMetadataSet.extract({"foo/primitive_int": IntMetadataValue(5)}).primitive_int == 5
+    assert (
+        MyMetadataSet.extract({"foo/primitive_float": FloatMetadataValue(5.0)}).primitive_float == 5
+    )
+    assert MyMetadataSet.extract(
+        {"foo/int_metadata_value": IntMetadataValue(5)}
+    ).int_metadata_value == IntMetadataValue(5)
+    with pytest.raises(ValidationError):
+        assert MyMetadataSet.extract(
+            {"foo/int_metadata_value": 5}
+        ).int_metadata_value == IntMetadataValue(5)
```

Comment:
Let's remove the assert so it's clear what fails here (Pydantic instantiation)

```suggestion
        MyMetadataSet.extract({"foo/int_metadata_value": 5})
```

Also, what's the error message that we're showing here?

Reply:
oops, copy pasta. I removed